### PR TITLE
fix(hermes): restore L3 persona prompt parity

### PR DESCRIPTION
**Pull request by dplush**
## Summary

- restores L3 persona prompt injection in the root `hermes_memory_provider` surface
- keeps the behavior opt-in via `MNEMOSYNE_PERSONA_ENABLED=true`
- adds provider parity coverage for `system_prompt_block()` persona injection

## Why

`mnemosyne_hermes` already appended the `# L3 Persona (Active Behavioral Rules)` block when an enabled, non-empty persona file was present. The root `hermes_memory_provider` package had the persona tools/schemas, but missed the prompt injection path. Some real installs still load that root provider package, so L3 appeared configured but never reached the system prompt.

## Tests

- `PYTHONPATH=".:integrations/hermes/src" uv run --no-sync --with pytest --with pyyaml pytest tests/test_hermes_provider_parity.py tests/test_persona_injection.py tests/test_provider_all_15_tools.py -q`
- `python3 -m py_compile hermes_memory_provider/__init__.py tests/test_hermes_provider_parity.py`
- manual provider smoke: both `hermes_memory_provider` and `mnemosyne_hermes` include Mnemosyne header, L3 header, and persona file content

## Summary by CodeRabbit

* **New Features**
  * Hermes-related prompts can now automatically include optional persona text from a markdown file, with header injection and token/size-based truncation.
* **Bug Fixes**
  * More robust handling when persona prompting is disabled, the persona file is missing, or configuration values are invalid—prompt composition now safely falls back without failing.
* **Tests**
  * Added parity and regression coverage to ensure persona prompt behavior matches across provider implementations, including edge cases like negative/invalid token-cap values and resilient imports in reduced environments.

### CODE REVIEW

**Comment by dplush** [comment]
@coderabbitai review

File: hermes_memory_provider/__init__.py [hunk 1]
```diff
@@ -3371,6 +3375,47 @@ def on_memory_write(self, action: str, target: str, content: str) -> None:
...
+                last_section = truncated.rfind("\n## ")
+                if last_section > 0:
+                    truncated = truncated[:last_section]
+                raw = truncated + "\n... (truncated, see persona file for full content)"
+            self._persona_cache = {"mtime": mtime, "content": raw}
+            return raw
+        except Exception as exc:
+            logger.debug("persona_block read failed: %s", exc)
+            return ""
+
```
[the remark this answers is not in the document]
Comment (dplush) [reply hermes_memory_provider/__init__.py]: Fixed in `9a30533`: persona prompt logic now lives in shared `mnemosyne.integrations.hermes_persona_prompt.HermesPersonaPromptMixin`, used by both Hermes provider surfaces. I also kept `mnemosyne_hermes` import resilient with a fallback mixin for installer/status diagnostics and bumped the `mnemosyne-memory` floor to the version that ships the helper.

File: hermes_memory_provider/__init__.py [hunk 2]
```diff
@@ -3371,6 +3375,47 @@ def on_memory_write(self, action: str, target: str, content: str) -> None:
...
+    # L3 persona file injection. Default OFF -- opt in via
+    # MNEMOSYNE_PERSONA_ENABLED=true. When OFF, no file IO happens.
+    PERSONA_ENABLED = _parse_env_bool("MNEMOSYNE_PERSONA_ENABLED", False)
+    PERSONA_FILE = Path(
+        os.environ.get(
+            "MNEMOSYNE_PERSONA_FILE",
+            str(Path.home() / ".hermes" / "memory" / "persona.md"),
+        )
+    )
+    PERSONA_TOKEN_CAP = int(os.environ.get("MNEMOSYNE_PERSONA_TOKEN_CAP", "1500"))
```
[the remark this answers is not in the document]
Comment (dplush) [reply hermes_memory_provider/__init__.py]: Fixed in `9a30533`: `MNEMOSYNE_PERSONA_TOKEN_CAP` now uses safe parsing in the shared helper and falls back to `1500` for invalid/empty values. Added parity coverage for the invalid-env case.

File: tests/test_hermes_provider_parity.py [hunk 3]
```diff
@@ -266,6 +266,54 @@ def test_provider_sync_turn_zero_limit_means_untruncated(monkeypatch, provider_m
...
+        provider = _prompt_provider(module)
+        provider.PERSONA_ENABLED = False
+        provider.PERSONA_FILE = missing_file
+        assert "# L3 Persona" not in provider.system_prompt_block()
+
+        provider = _prompt_provider(module)
+        provider.PERSONA_ENABLED = True
+        provider.PERSONA_FILE = missing_file
+        assert "# L3 Persona" not in provider.system_prompt_block()
+
```
[the remark this answers is not in the document]
Comment (dplush) [reply tests/test_hermes_provider_parity.py:692]: Fixed in `9a30533`: added coverage for `PERSONA_ENABLED = False` with an existing persona file, asserting both the L3 header and file content stay suppressed.

**Comment by dplush** [comment]
@coderabbitai review

File: integrations/hermes/src/mnemosyne_hermes/__init__.py [hunk 4]
```diff
@@ -27,9 +27,24 @@
...
+
+        PERSONA_ENABLED = False
+        PERSONA_FILE = Path.home() / ".hermes" / "memory" / "persona.md"
+        PERSONA_TOKEN_CAP = 1500
+
+        def _persona_block(self) -> str:
+            return ""
+
+        def _with_persona_block(self, base: str) -> str:
+            return base
```
[the remark this answers is not in the document]
Comment (dplush) [reply integrations/hermes/src/mnemosyne_hermes/__init__.py:102]: Fixed in `ab6a720`: the fallback now logs a warning that the L3 persona mixin is unavailable and points operators to upgrade `mnemosyne-memory`; installer/status import resilience is preserved.

File: mnemosyne/integrations/hermes_persona_prompt.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                last_section = truncated.rfind("\n## ")
+                if last_section > 0:
+                    truncated = truncated[:last_section]
+                raw = truncated + "\n... (truncated, see persona file for full content)"
+            self._persona_cache = {"mtime": mtime, "content": raw}
+            return raw
+        except Exception as exc:
+            logger.debug("persona_block read failed: %s", exc)
+            return ""
+
```
[the remark this answers is not in the document]
Comment (dplush) [reply mnemosyne/integrations/hermes_persona_prompt.py:91]: Fixed in `ab6a720`: `max_words` is now clamped to a non-negative value before slicing, and I added a regression test so a negative `PERSONA_TOKEN_CAP` cannot leak tail content from the persona file.

**Comment by dplush** [comment]
@coderabbitai review

**Comment by AxDSan** [comment]
This one has merge conflicts against the current main — a batch of PRs landed today including the batch tool, hermes config, and hygiene features that touch the same files. I resolved them locally but could't push to your fork branch. Could you rebase on the current main? Your PR content is solid and CodeRabbit approved it — just needs a clean merge base.

**Comment by dplush** [comment]
Rebased on current main and addressed the new CodeRabbit feedback.

CI is green again, CodeRabbit approved on the current head, and the PR is clean/mergeable now. Thanks!
